btree/btree.c

```diff
@@ -4706,9 +4705,11 @@ static int64_t btree_get_kv_tick(struct m0_sm_op *smop)
 				s.s_idx = bop->bo_opc == M0_BO_MINKEY ? 0 :
 					  count - 1;
 				node_rec(&s);
-			} else
+			} else {
 				/** Only root node is present and is empty. */
-				s.s_rec.r_flags = M0_BSC_KEY_NOT_FOUND;
+				lock_op_unlock(tree);
+				return fail(bop, M0_ERR(-ENOENT));
+			}
 		}
 
 		bop->bo_cb.c_act(&bop->bo_cb, &s.s_rec);
```

Comment:
set sm_rc flag as M0_BSC_SUCCESS , as in ut's we are adding ASSERT based rc == M0_BSC_SUCCESS , currently M0_BSC_SUCCESS code is 0 even if we do not set sm_rc to M0_BSC_SUCCESS , it won't fail but to avoid future changes you might want to set sm_rc to M0_BSC_SUCCESS if operation is successful.

Same thing need to be done in iter_tick routine.